index.js

```diff
@@ -78,7 +78,7 @@ Router.prototype.on = function (method, path, handler, store) {
 
       if (isRegex && (i === len || path.charCodeAt(i) === 47)) {
         nodeType = 3
-      } else if (i < len) {
+      } else if (i < len && path.charCodeAt(i) !== 47) {
```

Comment:
The missing of the check against `path.charCodeAt(i)` caused that every parametric part of the url that didn't contain a regular expression, and was followed by another part of the url, was treated as a multi-parametric part.